CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+  classDef Black fill:white,stroke:black
+  classDef Blue fill:white,stroke:blue
+  classDef Green fill:white,stroke:green
+
+  start:::White --> StIdle:::Blue;
+  StIdle:::Blue --> StBusy:::Green : MsgSubmitMessage
+  StBusy:::Green --> StIdle:::Blue : MsgAcceptMessage
+  StBusy:::Green --> StIdle:::Blue : MsgRejectMessage
+  StIdle:::Blue --> StDone:::Black : MsgDone
+   
+```
+
+#### Protocol messages
+
+* **MsgSubmitMessage(message)**: The client submits a message.
+* **MsgAcceptMessage**: The server accepts the message.
+* **MsgRejectMessage(reason)**: The server rejects the message and replies with a *reason*.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgSubmitMessage | message | StBusy |
+| StBusy | MsgAcceptMessage |  | StIdle |
+| StBusy | MsgRejectMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+
+## Local Message Notification mini-protocol
+
+### Description
+
+The local message notification mini-protocol is used by local clients to be notified about new message received by the core node. 
+
+The protocol follows a simple request-response pattern:
+
+1. The client sends a request with a single message.
+2. The server either accepts the message (and returns a confirmation) or rejects it (and returns the reason)
+
+### State machine
+
+| Agency ||
+|--------|------| 
+| Client has Agency | StIdle | 
+| Server has Agency | StBusy |
+
+```mermaid
+stateDiagram-v2   
+
+  classDef White fill:white,stroke:white
+  classDef Black fill:white,stroke:black
+  classDef Blue fill:white,stroke:blue
+  classDef Green fill:white,stroke:green
+
+  start:::White --> StIdle:::Blue;
+  StIdle:::Blue --> StBusy:::Green : MsgNextMessage
+  StBusy:::Green --> StIdle:::Blue : MsgHasMessage
+  StBusy:::Green --> StIdle:::Blue : MsgTimeoutMessage
+  StIdle:::Blue --> StDone:::Black : MsgDone
+   
+```
+
+#### Protocol messages
+
+* **MsgNextMessage**: The client asks for the next message.
+* **MsgHasMessage(message)**: The server has received a message.
+* **MsgTimeoutMessage**: The server has not received a message and times out.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgNextMessage |  | StBusy |
+| StBusy | MsgHasMessage | message | StIdle |
+| StBusy | MsgTimeoutMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+# Rationale
+
+## Why are we proposing this CIP?
+
+### For Mithril
+
+* Mithril requires strong network foundations to support interactions between its various nodes:
+
+  * Mithril needs to exist in a decentralized context where multiple aggregators can operate seamlessly and independantly.
+  * Mithril needs participation of all or nearly all of the Cardano network SPOs to provide maximal security to the multi-signatures embedded in the certificates.
```

Comment:
> all or nearly all

Just say "the majority"?

Reply:
It's a very large majority of at least 95%